parcels/particleset.py

```diff
@@ -490,7 +498,7 @@ def remove_indices(self, indices):
     def remove_booleanvector(self, indices):
         """Method to remove particles from the ParticleSet, based on an array of booleans"""
         for d in self.particle_data:
-            self.particle_data[d] = self.particle_data[d][~indices]
+            self.particle_data[d] = self.particle_data[d][..., ~indices]
```

Comment:
uncommon notation (thought this only works with ``...`` being at the end of a command) - nice that it works as simplification!